Description: Let CUPS stop on 30 seconds idle time if it was started on-demand
 by Upstart
Author: Till Kamppeter <till.kamppeter@gmail.com>,
Bug: https://bugs.launchpad.net/ubuntu/+source/cups/+bug/1276713
Last-Update: 2015-02-10
--- a/scheduler/main.c
+++ b/scheduler/main.c
@@ -1735,6 +1735,13 @@ upstart_checkin(void)
   if (httpAddrPort(&(lis->address)) == 443)
     lis->encryption = HTTP_ENCRYPT_ALWAYS;
 #  endif /* HAVE_SSL */
+#if defined(HAVE_LAUNCHD) || defined(HAVE_SYSTEMD)
+  /* As we are started on-demand, stop on idle */
+  if (!IdleExitTimeout)
+    IdleExitTimeout = 30;
+  cupsdLogMessage(CUPSD_LOG_DEBUG, "As we are starting on-demand (socket-triggered), activate exit-on-idle mode, timeout: %d seconds.",
+                      IdleExitTimeout);
+#endif
 }
 
 /*
